[bug] Missing support for newer compilers

Here's an example of [Conan from pip as of yesterday](https://github.com/johnmcfarlane/cnl_ci_images/blob/master/dockerfiles/1/base-20.04#L20) which only, apparently, added GCC 9.3 less than a month ago to [here](https://github.com/conan-io/conan/blob/725bd0cee4e53f35521aef7eeb61d4772c460d5e/conans/client/conf/__init__.py#L67) failing to recognize GCC 10 in a [CI job](https://app.circleci.com/pipelines/github/johnmcfarlane/cnl/195/workflows/4fc93745-4137-4e23-a0c9-d652b8b70f35/jobs/7790).

GCC 10 is technically still in development. But it's already available in Ubuntu 20.04 which has been out for a few days. And some developers start using the latest version of GCC, Clang, etc. for quite a while before it's officially stable. Why must there continually be a window of time where Conan is broken for the newest compiler?

Why must similar issues, [like this one for Clang-10](https://github.com/conan-io/conan/issues/5918) be posted periorically? The versioning for these compilers is quite consistent now. And even if you don't trust that there will be a Clang-17 or a GCC-42, you can at least see what's on the horizon with roadmaps for [GCC](https://gcc.gnu.org/develop.html) and have reasonable certainty that there will be a [Clang 11](https://clang.llvm.org/docs/ReleaseNotes.html). 

In short, please consider doing at least one of two things:

* Monitor compiler releases and be ahead of the game so that Conan isn't routinely out of date for short windows of time. Maybe schedule updates or put a watch on pages like [this one](https://releases.llvm.org/) and [this one](https://gcc.gnu.org/develop.html).
* Lead with the version numbers so that all versions listed for a compiler in the roadmap are listed in _settings.yml_ **before that compiler is released**. Maybe even anticipate a version of two ahead. Be brave and speculate that there will be a GCC-11 and GCC-10.1 for example! But definitely put Clang-11 in there.

### Environment Details (include every applicable attribute)
  * Operating System+version: Linx
  * Compiler+version: 10 (and all subsequent versions!)
  * Conan version: 1.24.1 and HEAD
  * Python version: 3

### Steps to reproduce (Include if Applicable)

Use any version of Conan. Look at _~/.conan/settings.yml_. Observe that it's missing GCC-10 which is already available to users of the latest version of Ubuntu.

### Logs (Executed commands with output) (Include/Attach if Applicable)

> -- Conan executing: /usr/local/bin/conan install . -s build_type=Release -s compiler=gcc -s compiler.version=10 -s compiler.libcxx=libstdc++11 -g=cmake --build=outdated
> ERROR: Invalid setting '10' is not a valid 'settings.compiler.version' value.
> Possible values are ['4.1', '4.4', '4.5', '4.6', '4.7', '4.8', '4.9', '5', '5.1', '5.2', '5.3', '5.4', '5.5', '6', '6.1', '6.2', '6.3', '6.4', '7', '7.1', '7.2', '7.3', '7.4', '8', '8.1', '8.2', '8.3', '9', '9.1', '9.2']
> Read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-invalid-setting"
> CMake Error at /tmp/cnl/conan.cmake:402 (message):
>   Conan install failed='1'
> Call Stack (most recent call first):
>   /tmp/cnl/conan.cmake:497 (conan_cmake_install)
>   test/cmake/install-conan.cmake:11 (conan_cmake_run)
>   test/CMakeLists.txt:1 (include)

--- a/conans/client/conf/__init__.py
+++ b/conans/client/conf/__init__.py
@@ -64,7 +64,8 @@ _t_default_settings_yml = Template(textwrap.dedent("""
                       "6", "6.1", "6.2", "6.3", "6.4",
                       "7", "7.1", "7.2", "7.3", "7.4",
                       "8", "8.1", "8.2", "8.3",
-                      "9", "9.1", "9.2", "9.3"]
+                      "9", "9.1", "9.2", "9.3",
+                      "10"]
             libcxx: [libstdc++, libstdc++11]
             threads: [None, posix, win32] #  Windows MinGW
             exception: [None, dwarf2, sjlj, seh] # Windows MinGW

